lib/Conversion/TorchToTcp/Elementwise.cpp

```diff
@@ -60,54 +60,144 @@ class ConvertAtenAddSubOp : public OpConversionPattern<AtenOpT> {
     RankedTensorType lhsType = lhs.getType().dyn_cast<RankedTensorType>();
 
     Value rhs = adaptor.getOther();
-    RankedTensorType rhsType = rhs.getType().dyn_cast<RankedTensorType>();
 
     RankedTensorType resultType =
         OpConversionPattern<AtenOpT>::getTypeConverter()
             ->convertType(op.getType())
             .template cast<RankedTensorType>();
 
-    if (!lhsType || !rhsType || !resultType)
+    if (!lhsType || !resultType)
       return rewriter.notifyMatchFailure(
           op, "Only Ranked Tensor types are supported in TCP");
 
-    lhs = torch_to_tcp::broadcastInLeadingDimsToMatchShape(rewriter, lhs, rhs);
-    rhs = torch_to_tcp::broadcastInLeadingDimsToMatchShape(rewriter, rhs, lhs);
+    auto inputAType = op.getSelf()
+                          .getType()
+                          .template dyn_cast<torch::Torch::ValueTensorType>()
+                          .getDtype();
+    auto outputType = op.getType()
+                          .template dyn_cast<torch::Torch::ValueTensorType>()
+                          .getDtype();
+
+    if (isa<AtenAddScalarOp>(op) || isa<AtenSubScalarOp>(op)) {
+      RankedTensorType tensorResultType =
+          RankedTensorType::get({}, adaptor.getOther().getType());
+      rhs = torch_to_tcp::scalarToTcpTensor(rewriter, op, tensorResultType,
+                                            op.getOther());
+      if (adaptor.getOther().getType().template isa<mlir::FloatType>())
+        // FP rhs is treated as fp64
```

Comment:
Why is this the case? What about f32 additions, for example?

Reply:
Looks like torch.constant.float will be treated as [64 bit floating point value](https://github.com/llvm/torch-mlir/blob/f6a6cfea4e4f71867127504c0315082e1eb8e354/include/torch-mlir/Dialect/Torch/IR/TorchOps.td#L741). So I just use fp64 as its type to perform the cast if necessary.